src/renderer/shared/lib/utils/assets.ts

```diff
@@ -1,36 +1,48 @@
-import { type ApiPromise } from '@polkadot/api';
-
-import { type Asset, AssetType, type OrmlExtras, StakingType, type StatemineExtras } from '@/shared/core/types/asset';
+import {
+  type Asset,
+  type AssetId,
+  AssetType,
+  type OrmlExtras,
+  StakingType,
+  type StatemineExtras,
+} from '@/shared/core/types/asset';
 import { assert } from '@/shared/lib/utils';
 
-export const getNativeAsset = (assets: Asset[]): Asset => {
-  const nativeAsset = assets.find((asset) => asset.type === AssetType.NATIVE);
-  if (!nativeAsset) {
-    // some networks use orml assets as native (cringe)
-    const firstAsset = assets.at(0);
-    assert(firstAsset, 'Native asset is not found');
-    return firstAsset;
-  }
+/**
+ * Asset id as seen by the chain. Useful when we want to transact specifying
+ * this asset. Note that this can have different type depending on the type of
+ * the corresponding asset
+ */
+export type OnChainAssetId = string | number | 'Native';
```

Comment:
Can someone confirm whether storing 'Native' for native in the db is fine in terms of backward compatibility?